projects/addon-mobile/directives/ripple/ripple.directive.ts

```diff
@@ -1,79 +1,158 @@
+    private readonly destroy$ = inject(DestroyRef);
     private readonly renderer = inject(Renderer2);
-    private readonly start$ = inject(TUI_RIPPLE_START);
-    private readonly end$ = inject(TUI_RIPPLE_END);
+    private readonly animationSpeed = inject(TUI_ANIMATIONS_SPEED);
 
-    @Input()
-    public tuiRipple?: string | '';
-
-    protected readonly nothing = tuiWithStyles(TuiRippleStylesComponent);
+    @Input({
+        alias: 'tuiRipple',
+        transform: selectorScopeFallback,
+    })
+    public selector = ':scope';
 
     constructor() {
-        const touchEnd$ = tuiTypedFromEvent(this.el, 'touchend');
-        const touchMove$ = tuiTypedFromEvent(this.el, 'touchmove');
+        tuiWithStyles(TuiRippleComponent);
+    }
+
+    @HostListener('pointerdown.silent', [
+        '$event.clientX',
+        '$event.clientY',
+        '$event.target',
+    ])
+    protected start(clientX: number, clientY: number, target: HTMLElement): void {
+        const element = target.closest(this.selector);
+
+        if (!(element instanceof HTMLElement)) {
+            return;
+        }
+
+        const ripple = this.createRipple(
+            clientX,
+            clientY,
+            element.getBoundingClientRect(),
+        );
 
-        this.end$.subscribe(element => this.renderer.removeChild(this.el, element));
+        this.renderer.addClass(element, 'tui-ripple-container');
 
-        this.start$
+        const touchEnd$ = merge(
+            tuiTypedFromEvent(element, 'touchend'),
+            tuiTypedFromEvent(element, 'mouseup'),
+            tuiTypedFromEvent(element, 'mouseout'),
+        );
+        const touchMove$ = merge(
+            tuiTypedFromEvent(element, 'touchmove'),
+            tuiTypedFromEvent(element, 'mousemove'),
+        );
+
+        race(TOUCH_MOVE_DELAY_TIMER, touchEnd$.pipe(map(TUI_TRUE_HANDLER)))
             .pipe(
-                mergeMap(ripple => {
-                    const animationEndOn$ = tuiTypedFromEvent(ripple, 'animationend');
-
-                    return race(
-                        timer(TOUCH_MOVE_DELAY).pipe(map(TUI_FALSE_HANDLER)),
-                        touchEnd$.pipe(map(TUI_TRUE_HANDLER)),
-                    ).pipe(
-                        take(1),
-                        // eslint-disable-next-line rxjs/no-unsafe-takeuntil
-                        takeUntil(touchMove$),
-                        tap(() => {
-                            this.renderer.setStyle(
-                                ripple,
-                                'background',
-                                this.tuiRipple || null,
-                            );
-                            this.renderer.appendChild(this.el, ripple);
+                take(1),
+                tap(() => {
+                    this.renderer.appendChild(element, ripple);
+                }),
+                switchMap(isTap => {
+                    const animation = ripple.animate(RIPPLE_ON_ANIMATION, {
```

Comment:
nitpicking: looks like we do not really need this constant, especially when we switch to keeping `this.duration`, like we do not have it later at line 124.